go/oasis-test-runner/cmd/cmp/cmp.go

```diff
@@ -674,6 +676,9 @@ func Register(parentCmd *cobra.Command) {
 	)
 	cmpFlags.String(cfgMetricsNetDevice, "lo", "network device traffic to compare")
 
+	cmpFlags.String(cfgMetricsAddr, "127.0.0.1:3000", "metrics pull address")
```

Comment:
This address flag overwrites the address flag in the `root.go` file on line `cmp.Register(rootCmd)` as they both share the same name. Therefore, condition `viper.IsSet(cfgMetricsAddr)` is never true and metrics don't work. If I change the name of the cmp address flag or don't register the cmp command, then metrics work as expected. 

Tested locally with the following command. 
`.buildkite/scripts/test_e2e.sh -s=e2e.runtime.keymanager-restart --metrics.address 127.0.0.1:9091`

Reply:
Ugh, it's this dumb viper bug again...  I'll make a PR with a workaround for this.

Reply:
Should be fixed in https://github.com/oasisprotocol/oasis-core/pull/5247 :)